catalog/spatial/registry/registry-identification-plugin/src/main/java/org/codice/ddf/registry/identification/RegistryIdPostIngestPlugin.java

```diff
@@ -0,0 +1,147 @@
+import ddf.catalog.operation.UpdateResponse;
+import ddf.catalog.operation.impl.QueryImpl;
+import ddf.catalog.operation.impl.QueryRequestImpl;
+import ddf.catalog.plugin.PluginExecutionException;
+import ddf.catalog.plugin.PostIngestPlugin;
+
+/**
+ * The RegistryIdPostIngestPlugin collects a set of registry-ids that are currently in the system
+ * and makes them available to the IdentificationPlugin for duplication checking.
+ *
+ * Initially the catalog is queried to initialize the set of registry-ids. After that the list is
+ * maintained by the process methods for create and delete responses.
+ */
+public class RegistryIdPostIngestPlugin implements PostIngestPlugin {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(IdentificationPlugin.class);
+
+    private static final int RETRY_INTERVAL = 30;
+
+    private ScheduledExecutorService executorService;
+
+    private CatalogFramework catalogFramework;
+
+    private FilterBuilder filterBuilder;
+
+    private Set<String> registryIds = ConcurrentHashMap.newKeySet();
+
+    public Set<String> getRegistryIds() {
+        return Collections.unmodifiableSet(registryIds);
+    }
+
+    @Override
+    public CreateResponse process(CreateResponse input) throws PluginExecutionException {
+        if (input != null && input.getCreatedMetacards() != null) {
+            input.getCreatedMetacards()
+                    .stream()
+                    .filter(RegistryUtility::isRegistryMetacard)
+                    .forEach(e -> registryIds.add(RegistryUtility.getRegistryId(e)));
+        }
+        return input;
+    }
+
+    @Override
+    public UpdateResponse process(UpdateResponse input) throws PluginExecutionException {
+        return input;
+    }
+
+    @Override
+    public DeleteResponse process(DeleteResponse input) throws PluginExecutionException {
+        if (input != null && input.getDeletedMetacards() != null) {
+            input.getDeletedMetacards()
+                    .stream()
+                    .filter(RegistryUtility::isRegistryMetacard)
+                    .forEach(e -> registryIds.remove(RegistryUtility.getRegistryId(e)));
+        }
+        return null;
+    }
+
+    public void init() {
+        try {
+            List<Metacard> registryMetacards;
+            Filter registryFilter = filterBuilder.attribute(Metacard.TAGS)
+                    .is()
+                    .like()
+                    .text(RegistryConstants.REGISTRY_TAG);
+            QueryImpl query = new QueryImpl(registryFilter);
+            query.setPageSize(1000);
+            QueryRequest request = new QueryRequestImpl(query);
+
+            QueryResponse response = Security.runAsAdminWithException(() -> catalogFramework.query(
+                    request));
+
+            if (response == null) {
+                throw new PluginExecutionException(
+                        "Plugin failed to initialize plugin, unable to query identity node.");
```

Comment:
Is this better `"Failed to initialize plugin. Query for registry metacards came back null"`